app/services/order_purchaser.rb

```diff
@@ -79,12 +79,19 @@ def quantities
     order.order_details.order("order_details.id").pluck(:quantity)
   end
 
-  def send_facility_notification?
-    order_notification_recipient.present? && !acting_as?
-  end
+  def send_purchase_notifications
+    if !acting_as? || params[:send_notification] == "1"
```

Comment:
I think I would prefer all these conditionals as separate private methods, for ease of reading and also more descriptiveness about what each section is doing.

i.e. `send_facility_notification_if_appropriate`, `send_product_specific_facility_notifaction_if_appropriate` etc.

Or maybe better, `send_facility_notification if should_send_facility_notification?`

Thoughts?